packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -141,10 +142,25 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
       } else {
         virtualizer.scrollToItem(focusedKey, {duration: 0});
       }
+
+      if (modality === 'keyboard' && ref.current.contains(document.activeElement)) {
+        // Wait till virtualizer scrolls the focused key into view before checking if we need to scroll it into the viewport
+        // Specifically need to do this for the column headers since they aren't part of the scrollable table body and their position
+        // syncing is a bit delayed compared to when scrollToItem is fired.
+        raf = requestAnimationFrame(() => {
```

Comment:
I know ScrollView has an onScrollEnd, is it possible to use that here instead

Reply:
Ah I hadn't considered this, definitely an intriguing idea since it reduces the randomness of when this raf executes w/ regards to the Virtualizer/ScrollView's scrolling. I tried it out but this raf actually needs to trigger even if the ScrollView doesn't perform a scroll, specifically for the case where the table itself is partially out of view.  

Ideally, what this would do is 
1. focusedKey changes
2. wait for the scrollview to scroll. If it doesn't need to scroll, then immediately proceed to the next step
3. check if the document.activeElement needs to be scrolled into view

Still thinking about a way to do step 2, maybe something in virtualizer state can give me that info

Reply:
Tried, using `virtualizer.getView` to check if the focused key is in view within the virtualizer, but it always returns as a non-null value inside the useEffect hrm...

Reply:
Some supplementary information: `onScrollStart` occur after this useEffect happens, meaning I cannot use onScrollStart as a way to set the global `delayScrolling` to delay scrolling until the column headers position themselves. Additionally, `virtualizer.visibleViews` includes more columns than are visually visible so `virtualizer.getView` won't help here. Gonna try using the visible rect of the virtualizer vs the layoutInfo of the focused key, might be promising

Reply:
Implemented in https://github.com/adobe/react-spectrum/pull/3943, but ran into some issues

Problematic cases:
- using the focusedKey's view's layoutInfo rect to check if it is partially out of view of the virtualizer's visible rect breaks if the focusedKey is the row itself.  The row might be very long and thus will always think that focusing it will need to cause a scroll in ScrollView
  -  Might be able to get around this by changing the intersection logic a bit, will need to think on this (perhaps if the width/height of the focused rect is greater than the visible rect, then we go with a block: nearest strategy and deem it only in need of scrolling if the top left corner isn't in view)

- relying on ScrollView's onScrollEnd is a bit delayed (can be seen by partially scrolling the bottom of the table out of view and then pressing arrow down to bring a row that isn't in view into view). Will need to find the earliest time possible where the column header position has finished scrolling to match the ScrollView's new position

Reply:
Pushed the change from above, lemme know what you think